### ISSUE TITLE
Incorrect container CPU Usage reported with 4.7 kernel

### ISSUE THREAD

**Issue by pgeorgiev333** [google/cadvisor#1676]
Original defect submitted in docker space but as per the comments i'm submitting it here:

https://github.com/moby/moby/issues/26591

Environment:

- Docker Engine 17.03.1-ce
- Kernel Version: 4.10.10-200.fc25.x86_64
- Operating System: Fedora 25 (Server Edition)
- cadvisor version: v0.25.0

Description:
caDvisor is started as a container using docker-compose:

```

cadvisor:
  restart: always
  image: google/cadvisor:v0.25.0
  container_name: cadvisor
  hostname: docker
  privileged: true
  environment:
  - TZ=Europe/Sofia
  volumes:
  - /cgroup:/cgroup:ro
  - /:/rootfs:ro
  - /var/run:/var/run:rw
  - /sys:/sys:ro
  - /var/lib/docker/:/var/lib/docker:ro
  command: -docker="tcp://<docker host IP>:2375"
  ports:
  - "7080:8080"
```

I use prometheus to gather the data. The host has 24 CPUS as per /proc/cpuinfo.

When i open the casvisor url with the metrics i see a lot more cpus with 0 usage (http://<docker host IP>:7080/metrics):

container_cpu_usage_seconds_total{cpu="cpu47",id="/system.slice/systemd-tmpfiles-setup-dev.service"} 0
container_cpu_usage_seconds_total{cpu="cpu47",id="/system.slice/systemd-tmpfiles-setup.service"} 0
container_cpu_usage_seconds_total{cpu="cpu47",id="/system.slice/systemd-udev-trigger.service"} 0
container_cpu_usage_seconds_total{cpu="cpu47",id="/system.slice/systemd-udevd.service"} 0
container_cpu_usage_seconds_total{cpu="cpu47",id="/system.slice/systemd-update-utmp.service"} 0
container_cpu_usage_seconds_total{cpu="cpu47",id="/system.slice/systemd-user-sessions.service"} 0
container_cpu_usage_seconds_total{cpu="cpu47",id="/system.slice/systemd-vconsole-setup.service"} 0
container_cpu_usage_seconds_total{cpu="cpu47",id="/user.slice"} 0
container_cpu_usage_seconds_total{cpu="cpu47",id="/user.slice/user-0.slice"} 0
container_cpu_usage_seconds_total{cpu="cpu47",id="/user.slice/user-0.slice/session-525520.scope"} 0
container_cpu_usage_seconds_total{cpu="cpu47",id="/user.slice/user-0.slice/session-567755.scope"} 0
container_cpu_usage_seconds_total{cpu="cpu47",id="/user.slice/user-0.slice/user@0.service"} 0

This messes up the statistics.

Expected behavior:
cAdvisor should report the correct number of CPUs.

Note that this is a bare-metal machine and not a VM!

Additional info:

**cat /sys/fs/cgroup/cpu/docker/545f6e6c2b32e45be640f392fdac2f6797437db1d04eec187fd61dd7f1c1866f/cpuacct.usage_percpu**
826118233179 58694372670 796609326607 56037464290 793973262202 54087045169 805818209148 56182990377 705803893553 55982369675 800119933601 54110921158 343507168085 21829856769 346579448581 20720299643 348909086400 19401350086 332530330101 19208140460 224405763829 21326978846 307281920151 21419385478 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0

**Comment by dashpole** [comment]
Thanks for this!  Ill try and fix this in the near future.

**Comment by fredsig** [comment]
Also seeing this on ECS instances running latest AMI:

I0725 15:13:55.075525   19818 manager.go:201] Version: {KernelVersion:4.9.27-14.33.amzn1.x86_64 ContainerOsVersion:Amazon Linux AMI 2017.03 DockerVersion:17.03.1-ce CadvisorVersion:v0.24.1 CadvisorRevision:ae6934c}

I have 2 cores, but seeing 14 instead:

container_cpu_usage_seconds_total{cpu="cpu00",id="/"} 135670.315078045
container_cpu_usage_seconds_total{cpu="cpu01",id="/"} 132425.496029752
container_cpu_usage_seconds_total{cpu="cpu02",id="/"} 0
container_cpu_usage_seconds_total{cpu="cpu03",id="/"} 0
container_cpu_usage_seconds_total{cpu="cpu04",id="/"} 0
container_cpu_usage_seconds_total{cpu="cpu05",id="/"} 0
container_cpu_usage_seconds_total{cpu="cpu06",id="/"} 0
container_cpu_usage_seconds_total{cpu="cpu07",id="/"} 0
container_cpu_usage_seconds_total{cpu="cpu08",id="/"} 0
container_cpu_usage_seconds_total{cpu="cpu09",id="/"} 0
container_cpu_usage_seconds_total{cpu="cpu10",id="/"} 0
container_cpu_usage_seconds_total{cpu="cpu11",id="/"} 0
container_cpu_usage_seconds_total{cpu="cpu12",id="/"} 0
container_cpu_usage_seconds_total{cpu="cpu13",id="/"} 0
container_cpu_usage_seconds_total{cpu="cpu14",id="/"} 0

**Comment by dashpole** [comment]
this should be fixed by #1711
